replay/models/lin_ucb.py

```diff
@@ -0,0 +1,473 @@
+import numpy as np
+import pandas as pd
+import os
+
+from typing import Any, Dict, List, Optional, Tuple
+
+from replay.data.dataset import Dataset
+from replay.metrics import NDCG, Metric
+from replay.utils import SparkDataFrame
+
+from .base_rec import HybridRecommender
+
+from replay.utils.spark_utils import convert2spark
+
+# import for parameter optimization
+from optuna import create_study
+from optuna.samplers import TPESampler
+
+from tqdm import tqdm
+import scipy.sparse as scs
+
+
+# Object for interactions with a single arm in a UCB disjoint framework
+class linucb_disjoint_arm:
+    def __init__(
+        self, arm_index, d, eps, alpha
+    ):  # in case of lin ucb with disjoint features: d = dimension of user's features solely
```

Comment:
This line might be misinterpreted. E.g. there might be an additional feature with constant 1, as in the original article (section 5.2.2)